### DISCUSSION TITLE
How to use winget to install Git with custom options?

### DISCUSSION THREAD

**Discussion by hakan-demirli** [microsoft/winget-cli#3462] [Q&A] [answered]
How can I toggle installation options for Git?
I have found a discussion which shows how to do it for vscode. I have tried similar namings for Git but no luck.
https://github.com/microsoft/winget-cli/discussions/1798

```winget install -e --id Git.Git --override '/SILENT /mergetasks="!runcode,gitbashhere,gitguihere"'```
```winget install -e --id Git.Git --override "/i Git InstallerFull.msi gitbashhere=0,gitguihere=0"'```
```winget install -e --id Git.Git --override "/installArgs:\"gitbashhere=0,gitguihere=0\""```
I want to disable these two:
![image](https://github.com/microsoft/winget-cli/assets/78746991/9ee27b0c-fe02-49c8-9f4a-d40ac257946d)

**Comment by byjrack** [comment, marked as answer]
Probably want to swing through https://github.com/git-for-windows/git/discussions/4423 I am guessing the arg you are looking for you can pass in using the /o param to Inno.  You can still use the inno feature to get the iss file for the command lines you want to use.  I then use --custom in winget to pass those to git for windows.  you can use --override, but you will need to also handle the visibility and logging args as well since those will be dropped.

**Comment by simshaun** [comment]
If anyone runs across this, I've managed to turn off options by explicitly declaring the components I want:

```console
winget install --id=Git.Git --custom '/components="gitlfs,assoc,autoupdate"'
```

List of components:  
https://github.com/git-for-windows/build-extra/blob/d1a9c4e920b476c4b4d439051ecddc9e281ca8a1/installer/install.iss#L107

Documentation on the /COMPONENTS parameter:  
https://jrsoftware.org/ishelp/index.php?topic=setupcmdline

I have not figured out how to change some other options like default editor, whether to use bundled/external SSH, line endings, etc. For now, seems easier just to use `--interactive`.

**Reply by rimrul** [reply]
What you're looking for is `/o:<key>=<value>`. You can find a list of [available options here](https://github.com/git-for-windows/git/wiki/Silent-or-Unattended-Installation#custom-installer-options) and an overview of how they [correspond to the installer GUI here](https://github.com/git-for-windows/git/wiki/Mapping-between-Git-Installer-GUI-Settings-And-Command-Line-Arguments).
